### PR TITLE
B-04007 use profiles zome

### PR DESCRIPTION
**Pull request by robbiecarlton**
Uses the profile zome to store and retrieve user handle.

Works with this branch of EC https://github.com/holochain/elemental-chat/pull/54

### CODE REVIEW

File: src/store/elementalChat.js [hunk 1]
```diff
@@ -104,14 +104,27 @@ export default {
...
-    initialize ({ state, dispatch }) {
+    initialize ({ dispatch }) {
       const currentChannelId = window.localStorage.getItem('currentChannelId')
       if (currentChannelId) {
         dispatch('joinChannel', currentChannelId)
       }
+      dispatch('initializeAgent')
+    },
+    initializeAgent ({ dispatch, rootState }) {
+      const tryToGetProfile = () => {
```
Comment (robbiecarlton) [inline src/store/elementalChat.js:120]: at some point we might want to create a zome call queue that waits for the conductor to connect, to solve this problem in a more general way.

File: src/App.vue [hunk 2]
```diff
@@ -128,17 +128,18 @@ export default {
...
       'errorMessage'
     ]),
+    ...mapGetters('elementalChat', ['updateProfileLoading']),
     shouldDisplayNickPrompt () {
       return (
         this.needsHandle &&
         !this.errorMessage &&
         !this.conductorDisconnected &&
         !this.shouldDisplayHoloConnecting
+        // !this.updateProfileLoading
```
Comment (timotree3) [inline src/App.vue:142]: commented code

File: src/ElementalChat.vue [hunk 3]
```diff
@@ -22,7 +22,7 @@
               icon
               v-bind="attrs"
               v-on="on"
-              @click="editHandle()"
+              @click="needsHandle()"
```
Comment (timotree3) [inline src/ElementalChat.vue:25]: Isn't needsHandle a bool?

File: src/store/elementalChat.js [hunk 4]
```diff
@@ -104,14 +104,27 @@ export default {
     channels: [],
     currentChannelId: null,
     stats: {},
-    statsLoading: false
+    statsLoading: false,
+    agentHandle: '',
+    needsHandle: false
```
Comment (timotree3) [inline src/store/elementalChat.js:109]: Possible clarifying comment if it's not possible to just get rid of this field
```suggestion
    // If needsHandle is true then we prompt the user to enter a handle
    needsHandle: false
```

File: src/store/elementalChat.js [hunk 5]
```diff
@@ -104,14 +104,27 @@ export default {
     channels: [],
     currentChannelId: null,
     stats: {},
-    statsLoading: false
+    statsLoading: false,
+    agentHandle: '',
```
Comment (timotree3) [inline src/store/elementalChat.js:108]: This should probably start out `null` given that when we reset it that's what we set it to
```suggestion
    agentHandle: null,
```

**Review by timotree3 — APPROVED** [review]
Looks good besides calling `needsHandle()`. I did find a bit hard to track what the logic was doing the first couple readings but now I get that `needsHandle` starts out false, and only when we try to get an existing profile and fail, we set it to true, prompting the user to enter a nickname.

Would it be possible to get rid of needsHandle and just check if the `agentHandle` is null?

If not it might help to just add a comment to the `needsHandle` definition stating what it means

File: src/ElementalChat.vue (hunk 3 above)
Comment (robbiecarlton) [reply src/ElementalChat.vue:25]: horribly, there is a `needsHandle` bool and a `needsHandle` mutation. I'm gonna fix that so it's easier to read

**Comment by robbiecarlton** [comment]
> Would it be possible to get rid of needsHandle and just check if the `agentHandle` is null?

What you lose with this approach is that if the user decides to update their handle, they're not presented with their current handle in the form input (because we had to set it to null to show the box).

> If not it might help to just add a comment to the `needsHandle` definition stating what it means
done

**Review by timotree3 — APPROVED** [review]

File: src/store/elementalChat.js [hunk 6]
```diff
@@ -278,10 +292,25 @@ export default {
...
       callZome(dispatch, rootState, 'chat', 'refresh_chatter', null, 30000)
-        .catch(error => log('refreshChatter zome error', error))
     },
     joinChannel ({ commit }, payload) {
       commit('setCurrentChannelId', payload)
+    },
+    updateProfile ({ commit, dispatch, rootState }, payload) {
+      const args = {
+        nickname: payload,
+        avatar_url: ''
```
Comment (zo-el) [inline src/store/elementalChat.js:302]: `avatar_url` is an option so you don't need to pass it. unless you would like it to be a placeholder for now.

**Review by zo-el — APPROVED** [review]
Do we need some basic unit-tests for this work? 
Also, I left one comment that is not a blocker.

**Comment by robbiecarlton** [comment]
> Do we need some basic unit-tests for this work?
Yes, I'll add a test.

**Pull request merged**